### DISCUSSION TITLE
warning it is strongly recommended to first generate an in silico spectral library in a separate pipeline step and then use it to process the raw data

### DISCUSSION THREAD

**Discussion by grandrea** [vdemichev/DiaNN#1074] [General]
Hello,

I ran DIA-NN 1.9 with MBR and in silico spectral library. I don't fully understand what this warning refers to and what workflow i should have used instead.

warning it is strongly recommended to first generate an in silico spectral library in a separate pipeline step and then use it to process the raw data now without activating FASTA digest

![diann](https://github.com/vdemichev/DiaNN/assets/44289027/93c12cbf-6ec4-401c-9b68-8ab2820fea09)

**Comment by vdemichev** [comment]
In the log, you see that DIA-NN saved a .predicted.speclib library.
What you can do now, is reuse this library for all similar analyses. So just put it in the 'Spectral library' field, uncheck 'FASTA digest' and uncheck 'Deep learning' - then save these settings as a separate pipeline step and run it.

Best,
Vadim

**Reply by sooheon** [reply]
Can you define "similar"? Any new samples, even of the same "type" should trigger new speclib, or would you reuse speclibs across mass spec runs / batches?

**Reply by vdemichev** [reply]
Any samples for which the settings used to generate .predicted.speclib are also suitablew

**Reply by sooheon** [reply]
So to be clear, this can be totally different experiment samples, as long as the settings remain the same?

**Comment by grandrea** [comment]
thanks! But the analysis I have already done on the very same data (library free, double pass etc with spectral library export) is fine, right? or it is expressely suggesting to re-run now with the spectral library?

**Reply by vdemichev** [reply]
Yes, it’s fine

**Comment by NathanielakaJuan** [comment]
Additional question to this. Why is it strongly recommended ? Does it change anything in case of output ? Is it only about speed ?

**Reply by vdemichev** [reply]
Better in terms if reproducibility & keeping track of what's going on. There's just no reason to do lib-free search on-the-fly when obtaining 'final analysis' for publication.

**Reply by vdemichev** [reply]
Yes, it changes the output.

**Reply by NathanielakaJuan** [reply]
Got it. Then I will do it this way also. Thank you!
